app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php

```diff
@@ -438,6 +438,10 @@ public function addAttributeToSelect($attribute, $joinType = false)
             } else {
                 $attrInstance = Mage::getSingleton('eav/config')
                     ->getAttribute($this->getEntity()->getType(), $attribute);
+                // failure to resolve an attribute from the eav config implies it does not exist
+                if (empty($attrInstance)) {
```

Comment:
Depending on issue #3057, may be need to check for `!$attrInstance->getId()` too.

Reply:
Yes, just waiting for 2nd comment in #3057 but I also agree that we probably should just return an empty model. Once we have a 3rd opinion I will create another PR.

Reply:
@davidhiendl this has beed addressed and we can consider it closed?

Reply:
@fballiano It will be addressed in the separate PR that will fix #3057 as it does not really belong here but in that PR instead.